common/logger/src/struct_log.rs

```diff
@@ -41,21 +43,30 @@ impl TcpWriter {
 
     /// Ensure that we get a connection, no matter how long it takes
     /// This will block until there is a connection
-    fn refresh_connection(&mut self) {
-        loop {
+    fn refresh_connection(&mut self) -> io::Result<()> {
+        // Only refresh the connection once a second
+        if self
+            .last_connection_attempt
+            .map(|t| t.elapsed() > Duration::from_millis(1000))
+            .unwrap_or(true)
+        {
+            self.last_connection_attempt = Some(Instant::now());
```

Comment:
So, here's the only problem.

The reason for blocking on the refreshing connection was so that we could buffer logs in the event of an outage.  So, now we drop logs immediately if there's an outage, and the buffer is only for if too many logs come in at once.

There's nothing wrong with that, except that it means we will drop as many logs as we can in a second between reconnects.  So, say you restart the logging node, you'll start dropping logs until it starts up again rather than buffering them.